x-pack/plugins/observability_solution/infra/public/components/ml/anomaly_detection/anomalies_table/anomalies_table.tsx

```diff
@@ -134,13 +152,54 @@ const AnomalyActionMenu = ({
   ];
 
   if (!disableShowInInventory) {
-    items.push(
-      <EuiContextMenuItem key="showInInventory" icon="search" onClick={showInInventory}>
+    const buildFilter = buildCombinedAssetFilter({
+      field: HOST_NAME_FIELD,
+      values: influencers,
+      dataView: metricsView?.dataViewReference,
+    });
+
+    let newFilter: Filter[] = [];
+    if (!Array.isArray(buildFilter)) {
+      newFilter = [buildFilter];
+    }
+
+    const showInHostsItem = !hostName ? (
+      <EuiContextMenuItem
+        key="showAffectedHosts"
+        icon="search"
+        data-test-subj="infraAnomalyFlyoutShowAffectedHosts"
+        href={hostsLocator?.getRedirectUrl({
+          dateRange: {
+            from: timeRange.start,
+            to: timeRange.end,
+          },
+          filters: newFilter,
+        })}
+      >
         <FormattedMessage
-          id="xpack.infra.ml.anomalyFlyout.actions.showInInventory"
-          defaultMessage="Show in Inventory"
+          id="xpack.infra.ml.anomalyFlyout.actions.showAffectedHosts"
+          defaultMessage="Show affected Hosts"
         />
       </EuiContextMenuItem>
+    ) : (
+      <></>
+    );
+
+    items.push(
+      influencerField === HOST_NAME_FIELD ? (
+        showInHostsItem
+      ) : (
+        <EuiContextMenuItem
+          icon="search"
+          data-test-subj="infraAnomalyFlyoutShowInInventory"
+          onClick={showInInventory}
+        >
+          <FormattedMessage
+            id="xpack.infra.ml.anomalyFlyout.actions.showInInventory"
+            defaultMessage="Show in Inventory"
+          />
+        </EuiContextMenuItem>
+      )
```

Comment:
Sorry if I missed the discussion here but in case of hostName should we default to inventory or we don't want the button at all? I mean both can be correct it depends on what we want to have here.
You can ignore the suggestion if that was discussed already.
We can do something like:
```suggestion
    items.push(
      influencerField === HOST_NAME_FIELD && !hostName ? (
        <EuiContextMenuItem
          key="showAffectedHosts"
          icon="search"
          data-test-subj="infraAnomalyFlyoutShowAffectedHosts"
          href={hostsLocator?.getRedirectUrl({
            dateRange: {
              from: timeRange.start,
              to: timeRange.end,
            },
            filters: newFilter,
          })}
        >
          <FormattedMessage
            id="xpack.infra.ml.anomalyFlyout.actions.showAffectedHosts"
            defaultMessage="Show affected Hosts"
          />
        </EuiContextMenuItem>
      ) : (
        <EuiContextMenuItem
          icon="search"
          data-test-subj="infraAnomalyFlyoutShowInInventory"
          onClick={showInInventory}
        >
          <FormattedMessage
            id="xpack.infra.ml.anomalyFlyout.actions.showInInventory"
            defaultMessage="Show in Inventory"
          />
        </EuiContextMenuItem>
      )
    );
  }
```

Reply:
We decided to remove button when anomaly table is displayed in assets view or host fly out. Therefore, it needs to stay as it is. Thx!

Reply:
Ok, that makes sense! Thanks for clarifying :)